app/src/main/java/arjun/hanabiandroid/game/InitialState.java

```diff
@@ -0,0 +1,38 @@
+package arjun.hanabiandroid.game;
+
+import android.support.annotation.IntDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public class InitialState {
+    @Retention(RetentionPolicy.SOURCE)
+    @IntDef({
+            GAME_TYPE_NO_RAINBOW,
+            GAME_TYPE_RAINBOW_SUIT,
+            GAME_TYPE_RAINBOW_WILDCARD,
+    })
+    @interface GameType {
+    }
+
+    public static final int GAME_TYPE_NO_RAINBOW = 0;
+    public static final int GAME_TYPE_RAINBOW_SUIT = 1;
+    public static final int GAME_TYPE_RAINBOW_WILDCARD = 2;
+
+    @GameType
+    final int gameType;
+    final long seed;
+    final int numPlayers;
+    final int turnOrder;
```

Comment:
Yep, that's roughly what I had in mind. The server can just determine `turnOrder` based on the order of calls to `/joinGame?id=xxx`.